core/inspector.rs

```diff
@@ -109,13 +109,24 @@ impl v8::inspector::V8InspectorClientImpl for JsRuntimeInspector {
     &mut self.v8_inspector_client
   }
 
+  /// This method id called when a breakpoint is triggered, eg. using `debugger` statement. In that case
+  /// inspector sends `Debugger.paused` notification. Nested message loop should be run and process all
+  /// sent protocol commands until `quit_message_loop_on_pause` is called. After that execution will
+  /// return to inspector and then JavaScript execution will resume.
   fn run_message_loop_on_pause(&mut self, context_group_id: i32) {
+    eprintln!("run message loop on pause");
     assert_eq!(context_group_id, JsRuntimeInspector::CONTEXT_GROUP_ID);
     self.flags.borrow_mut().on_pause = true;
-    let _ = self.poll_sessions(None);
+    let r = self.poll_sessions(None);
+    eprintln!("finished run message loop on pause {:#?}", r);
+    assert!(
+      !self.flags.borrow().on_pause, 
+      "V8InspectorClientImpl::run_message_loop_on_pause returned before quit_message_loop_on_pause was called"
+    );
```

Comment:
This is root of the problem that this PR is meant to fix. This method is called when we've already called `poll_sessions()` (from `JsRuntime::poll_event_loop`) and a call to `poll_sessions()` here returns `BorrowMutError`. This method shouldn't return until `quit_message_loop_on_pause` is called otherwise inspector goes into broken state with all message handling being delayed by at least one message received.